src/components/ThemeProvider.tsx

```diff
@@ -0,0 +1,9 @@
+"use client";
+
+import * as React from "react";
+import { ThemeProvider as NextThemesProvider } from "next-themes";
+import { type ThemeProviderProps } from "next-themes/dist/types";
+
+export function ThemeProvider({ children, ...props }: ThemeProviderProps) {
```

Comment:
Doteraz si mal skoro všetky komponenty ako const, tu máš zrazu function. Je lepšie to dodržovat všade rovnako, lepšie to vyzerá.